solr/core/src/java/org/apache/solr/handler/loader/JsonLoader.java

```diff
@@ -556,82 +556,105 @@ private void parseFieldValue(SolrInputField sif) throws IOException {
       if (ev == JSONParser.OBJECT_START) {
         parseExtendedFieldValue(sif, ev);
       } else {
-        Object val = parseNormalFieldValue(ev, sif.getName());
+        Object val = parseNormalFieldValue(ev, sif);
         sif.setValue(val);
       }
     }
 
+    private Map<String, Object> generateExtendedValueMap(int ev) throws IOException {
```

Comment:
Can you please sequence the methods so this follows where it's called from so readers needn't look back up?